src/errors.ts

```diff
@@ -0,0 +1,55 @@
+import type { Session } from './interfaces.js';
+import { decodeJwt } from './jwt.js';
+
+export class AuthKitError extends Error {
+  data?: Record<string, unknown>;
+
+  constructor(message: string, cause?: unknown, data?: Record<string, unknown>) {
+    super(message);
+    this.name = 'AuthKitError';
+    this.cause = cause;
+    this.data = data;
+  }
+}
+
+export interface TokenRefreshErrorContext {
+  userId?: string;
+  sessionId?: string;
+}
+
+export class TokenRefreshError extends AuthKitError {
+  readonly userId?: string;
+  readonly sessionId?: string;
+
+  constructor(message: string, cause?: unknown, context?: TokenRefreshErrorContext) {
+    super(message, cause);
+    this.name = 'TokenRefreshError';
+    this.userId = context?.userId;
+    this.sessionId = context?.sessionId;
+  }
+}
+
+export function getSessionErrorContext(session?: Session | null): TokenRefreshErrorContext {
+  if (!session) {
+    return {};
+  }
+
+  const context: TokenRefreshErrorContext = {};
+
+  if (session.user?.id) {
+    context.userId = session.user.id;
+  }
+
+  if (session.accessToken) {
+    try {
+      const { payload } = decodeJwt(session.accessToken);
+      if (payload.sid) {
```

Comment:
nit: i think we should pull the user_id (`sub`) from the access token here too (instead of from the session above). i agree the session.user.id _should_ always be the same but since we already have the access token decoded here it makes sense to me to use that as the source of truth).